templates/editConn.js

```diff
@@ -50,20 +78,29 @@ class Actions {
       "connHost",
       "Host URI is empty for the connection"
     );
-    let connToken = this.validTextInput(
-      "connToken",
-      "Token is empty for the connection",
-      "^[-A-Za-z0-9+=]{1,50}|=[^=]|={3,}$",
-      "Token should be a valid base64 encoded string"
-    );
-    let connOrg = this.validTextInput(
-      "connOrg",
-      "Org is empty for the connection"
-    );
+    let connToken, connOrg;
+    let connVersion = document.getElementById("connVersion").value;
+    if (connVersion == 1) {
+      connToken = "";
+      connOrg = "";
+    } else {
+      connToken = this.validTextInput(
+        "connToken",
+        "Token is empty for the connection",
+        "^[-A-Za-z0-9+=]{1,50}|=[^=]|={3,}$",
+        "Token should be a valid base64 encoded string"
+      );
+      connOrg = this.validTextInput(
```

Comment:
one little nit here, could validTextInput be a standalone function that returns a list of errors?

Reply:
I think it's a standalone function before. Now wrapped as a method instead of a global hasErr. But it's just a small script only one page, so should be enough for now.